packages/cli/test/integration/shared/testDb.ts

```diff
@@ -0,0 +1,389 @@
+			);
+			process.exit(1);
+		}
+
+		const testDbName = `n8n_test_pg_${Date.now()}`;
+		await bootstrapPostgres.query(`CREATE DATABASE ${testDbName};`);
+
+		await Db.init(getPostgresOptions({ name: testDbName }));
+
+		return { testDbName };
+	}
+
+	if (dbType === 'mysqldb') {
+		const bootstrapMysql = await createConnection(getBootstrapMySqlOptions());
+
+		const testDbName = `n8n_test_mysql_${Date.now()}`;
+		await bootstrapMysql.query(`CREATE DATABASE ${testDbName};`);
+
+		await Db.init(getMySqlOptions({ name: testDbName }));
+
+		return { testDbName };
+	}
+
+	throw new Error(`Unrecognized DB type: ${dbType}`);
+}
+
+/**
+ * Drop test DB, closing bootstrap connection if existing.
+ */
+export async function terminate(testDbName: string) {
+	const dbType = config.get('database.type') as DatabaseType;
+
+	if (dbType === 'sqlite') {
+		await getConnection(testDbName).close();
+	}
+
+	if (dbType === 'postgresdb') {
+		await getConnection(testDbName).close();
+
+		const bootstrapPostgres = getConnection(BOOTSTRAP_POSTGRES_CONNECTION_NAME);
+		await bootstrapPostgres.query(`DROP DATABASE ${testDbName}`);
+		await bootstrapPostgres.close();
+	}
+
+	if (dbType === 'mysqldb') {
+		await getConnection(testDbName).close();
+
+		const bootstrapMySql = getConnection(BOOTSTRAP_MYSQL_CONNECTION_NAME);
+		await bootstrapMySql.query(`DROP DATABASE ${testDbName}`);
+		await bootstrapMySql.close();
+	}
+}
+
+/**
+ * Truncate DB tables for specified entities.
+ *
+ * @param entities Array of entity names whose tables to truncate.
+ * @param testDbName Name of the test DB to truncate tables in.
+ */
+export async function truncate(entities: Array<keyof IDatabaseCollections>, testDbName: string) {
+	const dbType = config.get('database.type');
+
+	if (dbType === 'sqlite') {
+		const testDb = getConnection(testDbName);
+		await testDb.query('PRAGMA foreign_keys=OFF');
+		await Promise.all(entities.map((entity) => Db.collections[entity]!.clear()));
+		return testDb.query('PRAGMA foreign_keys=ON');
+	}
+
+	if (dbType === 'postgresdb') {
+		const map: { [K in keyof IDatabaseCollections]: string } = {
+			Credentials: 'credentials_entity',
+			Workflow: 'workflow_entity',
+			Execution: 'execution_entity',
+			Tag: 'tag_entity',
+			Webhook: 'webhook_entity',
+			Role: 'role',
+			User: 'user',
+			SharedCredentials: 'shared_credentials',
+			SharedWorkflow: 'shared_workflow',
+			Settings: 'settings',
+		};
+
+		return Promise.all(
+			entities.map((entity) =>
+				getConnection(testDbName).query(
+					`TRUNCATE TABLE "${map[entity]}" RESTART IDENTITY CASCADE;`,
+				),
+			),
+		);
+	}
+
+	// MySQL truncation requires globals, which cannot be safely manipulated by parallel tests
+	if (dbType === 'mysqldb') {
+		await Promise.all(
+			entities.map(async (entity) => {
+				await Db.collections[entity]!.delete({});
```

Comment:
it shouldn't matter for tests, but do we also want to reset the auto-increment counter to stay consistent with the truncating dbs?

Reply:
Done.